scripts/gen_syscalls.py

```diff
@@ -151,7 +151,7 @@ def need_split(argtype):
 def union_decl(type):
     return "union { struct { uintptr_t lo, hi; } split; %s val; }" % type
 
-def wrapper_defs(func_name, func_type, args):
+def wrapper_defs(func_name, func_type, args, attr):
```

Comment:
Hm... thinking more on this it might make more sense to just rename the system call to z_current_get() or whatever, and then just define `k_current_get()` as an inline wrapper with the appropriate gcc attribute set.  Mucking with a code generator just to set an optimization parameter when its output gets used elsewhere seems like needless complexity...